pkg/nodeconfig/payload/payload.go

```diff
@@ -110,6 +110,36 @@ param(
     }
   }
 
+# This retries getting the HNS-network until it succeeds
+function Retry-GetHnsNetwork {
+     param (
+         [int]$retryCount = 20,
+         [int]$retryDelay = 10
+     )
+ 
+     $attempt = 0
+ 
+     while ($attempt -lt $retryCount) {
+         try {
+             $hns_network = Get-HnsNetwork | Where-Object { $_.Name -eq 'HNS_NETWORK' }
+             
+             # Check if hns_network is null
+             if ($null -eq $hns_network) {
+                 Write-Host "Attempt $($attempt + 1) returned null. Retrying..."
```

Comment:
```suggestion
     $attempt = 1
 
     while ($attempt -le $retryCount) {
         try {
             $hns_network = Get-HnsNetwork | Where-Object { $_.Name -eq 'HNS_NETWORK' }
             
             # Check if hns_network is null
             if ($null -eq $hns_network) {
                 Write-Host "Attempt $attempt returned null. Retrying..."
```

Consider starting the counter at `1` to avoid the arithmetic operation on each iteration.